### PR TITLE
Kafka: 2.0.0 -> 2.0.1, Add 2.1

### PR DESCRIPTION
**Pull request by NeQuissimus**
###### Motivation for this change
Updates

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by NeQuissimus** [comment]
@GrahamcOfBorg test kafka

**Comment by rasendubi** [comment]
@GrahamcOfBorg build apacheKafka_2_0 apacheKafka_2_1

File: nixos/tests/kafka.nix [hunk 1]
```diff
@@ -70,4 +70,6 @@ in with pkgs; {
   kafka_0_11 = makeKafkaTest "kafka_0_11" apacheKafka_0_11;
   kafka_1_0  = makeKafkaTest "kafka_1_0"  apacheKafka_1_0;
   kafka_1_1  = makeKafkaTest "kafka_1_1"  apacheKafka_1_1;
+  kafka_2_0  = makeKafkaTest "kafka_2_0"  apacheKafka_2_0;
+  kafka_2_1  = makeKafkaTest "kafka_2_1"  apacheKafka_2_1;
```
Comment (Mic92) [inline nixos/tests/kafka.nix]: It seems Kafka versions older then 0.11 should be removed: https://kafka.apache.org/downloads

Comment (Mic92) [reply nixos/tests/kafka.nix]: And some version also could use an upgrade.

**Pull request merged**